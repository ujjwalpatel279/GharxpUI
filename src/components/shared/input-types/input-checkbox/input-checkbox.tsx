import React, { ChangeEvent, forwardRef } from 'react';
import './input-checkbox.scss';

interface InputCheckboxProps {
  id: string;
  defaultValue?: boolean;
  onToggle?: (e: ChangeEvent<HTMLInputElement>) => void;
  dataValue?: string | number;
}

type Ref = HTMLInputElement;

export const InputCheckbox = forwardRef<Ref, InputCheckboxProps>(({ id, defaultValue, onToggle, dataValue }, ref) => {
  return (
    <div className="c-input-checkbox">
      <input
        id={id}
        type="checkbox"
        ref={ref}
        {...(dataValue && { 'data-value': dataValue })}
        defaultChecked={defaultValue}
        onChange={onToggle}
      />
    </div>
  );
});

InputCheckbox.displayName = 'InputCheckbox';
