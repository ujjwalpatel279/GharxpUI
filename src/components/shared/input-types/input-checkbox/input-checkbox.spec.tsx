import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { InputCheckbox } from './input-checkbox';

describe('Input checkbox Component', () => {
  afterEach(cleanup);

  it('renders the checkbox component', () => {
    const { container } = render(<InputCheckbox id="inputCheckbox" ref={() => null} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
