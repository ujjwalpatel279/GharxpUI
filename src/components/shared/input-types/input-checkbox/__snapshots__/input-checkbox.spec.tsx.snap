// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input checkbox Component renders the checkbox component 1`] = `
<div
  class="c-input-checkbox"
>
  <input
    id="inputCheckbox"
    type="checkbox"
  />
</div>
`;
